core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/SSHAction.java

```diff
@@ -61,12 +62,20 @@ public SSHAction(SSHActionConfig config) {
 
   @Override
   public void run(final ActionContext context) throws Exception {
+    // now that macros have been substituted, try validation again
+    config.validate();
+
     Connection connection = new Connection(config.host);
     try {
       connection.connect();
 
-      String password = Strings.isNullOrEmpty(config.password) ? null : config.password;
-      if (!connection.authenticateWithPublicKey(config.user, new File(config.privateKeyFile), password)) {
+      // get private key from secure store using user-specified key
+      char[] privateKey =
+        Bytes.toString(context.getSecureData(context.getNamespace(), config.privateKeyLookupKey).get())
+          .toCharArray();
+      String passphrase = Strings.isNullOrEmpty(config.passphrase) ? "" : config.passphrase;
+
+      if (!connection.authenticateWithPublicKey(config.user, privateKey, passphrase)) {
```

Comment:
Nope, it's just poorly named:

@param pemPrivateKey A <code>char[]</code> containing a DSA or RSA private key of the user in OpenSSH key format (PEM, you can't miss the "-----BEGIN DSA PRIVATE KEY-----" or "-----BEGIN RSA PRIVATE KEY-----" tag). The char array may contain linebreaks/linefeeds.
```
public synchronized boolean authenticateWithPublicKey(String user, char[] pemPrivateKey, String password);
```